docs/source/developers/java/building.rst

```diff
@@ -0,0 +1,308 @@
+    [INFO] Arrow Memory - Netty ............................... SUCCESS [  3.334 s]
+    [INFO] Arrow Vectors ...................................... SUCCESS [ 21.791 s]
+    [INFO] Arrow Compression .................................. SUCCESS [  3.854 s]
+    [INFO] Arrow Tools ........................................ SUCCESS [  8.359 s]
+    [INFO] Arrow JDBC Adapter ................................. SUCCESS [  8.847 s]
+    [INFO] Arrow Plasma Client ................................ SUCCESS [  2.459 s]
+    [INFO] Arrow Flight ....................................... SUCCESS [  2.357 s]
+    [INFO] Arrow Flight Core .................................. SUCCESS [ 38.837 s]
+    [INFO] Arrow Flight GRPC .................................. SUCCESS [  5.955 s]
+    [INFO] Arrow Flight SQL ................................... SUCCESS [ 17.390 s]
+    [INFO] Arrow Flight Integration Tests ..................... SUCCESS [  6.148 s]
+    [INFO] Arrow AVRO Adapter ................................. SUCCESS [  9.635 s]
+    [INFO] Arrow Algorithms ................................... SUCCESS [ 26.949 s]
+    [INFO] Arrow Performance Benchmarks ....................... SUCCESS [  4.211 s]
+    [INFO] Arrow Orc Adapter .................................. SUCCESS [  6.522 s]
+    [INFO] Arrow Gandiva ...................................... SUCCESS [01:20 min]
+    [INFO] Arrow Java Dataset ................................. SUCCESS [ 12.949 s]
+    [INFO] ------------------------------------------------------------------------
+    [INFO] BUILD SUCCESS
+    [INFO] ------------------------------------------------------------------------
+
+IDE Configuration
+=================
+
+Go to open java project and select java folder.
+
+This is the initial view for java project loaded with default profiles:
+
+.. image:: img/java_welcome.png
+
+Let's create our maven configuration: java-dataset
+
+.. image:: img/java_mvn_configuration.png
+
+Let's define our JRE runner:
+
+.. image:: img/java_jre_runner.png
+
+Let's configure additional environment properties:
+
+.. image:: img/java_jre_env_properties.png
+
+Let's run our java-dataset maven configuration:
+
+.. image:: img/java_run_mvn_configuration.png
+
+.. image:: img/java_run_mvn_configuration_result.png
+
+Common Errors
+=============
+
+1.- Errors related to:
+    - Could NOT find Boost (missing: Boost_INCLUDE_DIR system filesystem)
+    - Could NOT find Lz4 (missing: LZ4_LIB)
+    - Could NOT find zstd (missing: ZSTD_LIB)
+
+    Consider to configure parameters that download software at the build time with:
+
+.. code-block::
+
+    -Dre2_SOURCE=BUNDLED \
+    -DBoost_SOURCE=BUNDLED \
+    -Dutf8proc_SOURCE=BUNDLED \
+    -DSnappy_SOURCE=BUNDLED \
+    -DORC_SOURCE=BUNDLED \
+    -DZLIB_SOURCE=BUNDLED \
+
+2.- Errors related to:
+    - Caused by: java.lang.NoSuchFieldException: reservedMemory.
+
+    We could move from classBits.getDeclaredField("reservedMemory") to classBits.getDeclaredField("RESERVED_MEMORY")
```

Comment:
Remove this section and just file a JIRA

Reply:
Deleted

Reply:
What was the JIRA?

Reply:
https://issues.apache.org/jira/browse/ARROW-15834